/*
 *  Copyright (c) 2016, https://github.com/wubenqi
 *  All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
package zproto;

option java_package = "com.zchat.engine.zproto";
option java_outer_classname = "ZProtoPrivacy";
option optimize_for = CODE_SIZE;

import "peers.proto";

// Privacy

// Block User
// BlockUserReq --> SeqRsp
message BlockUserReq {
  // int32 COMMAND_ID = 2636;
  UserOutPeer peer = 1; // Peer for blocking
}

// SeqRsp

// Unblock User
// UnblockUserReq --> SeqRsp
message UnblockUserReq {
  // int32 COMMAND_ID = 2637;
  UserOutPeer peer = 1; // Peer for unblocking
}

// SeqRsp

// Load Blocked Users
// LoadBlockedUsersReq --> LoadBlockedUsersRsp
message LoadBlockedUsersReq {
  // int32 COMMAND_ID = 2638;
}

message LoadBlockedUsersRsp {
  // int32 COMMAND_ID = 2639;
  repeated UserOutPeer user_peers = 1; //   Blocked user peers
}

// Update about User Blocked
message UserBlockedNotify {
  // int32 COMMAND_ID = 2629;
  int32 uid = 1; //   User Id
}

// Update about User Unblocked
message UserUnblockedNotify {
  // int32 COMMAND_ID = 2630;
  int32 uid = 1; //   User Id
}

